base/threadingconstructs.jl

```diff
@@ -38,8 +38,7 @@ function _threadsfor(iter,lbody)
                 end
             end
             # run this thread's iterations
-            for i = f:l
-                local $(esc(lidx)) = Base.unsafe_getindex(r,i)
+            for $(esc(lidx)) = r[f:l]
```

Comment:
Then what about using `view` instead of `getindex`? 

It seems to work with `Range`s as it should. That it doesn't work with `CartesianRange` is not its fault, but of CR.